geolocator_android/android/src/main/java/com/baseflow/geolocator/location/LocationManagerClient.java

```diff
@@ -73,45 +75,35 @@ static boolean isBetterLocation(Location location, Location bestLocation) {
     return false;
   }
 
-  private static String getBestProvider(
-      LocationManager locationManager, LocationAccuracy accuracy) {
-    Criteria criteria = new Criteria();
-
-    criteria.setBearingRequired(false);
-    criteria.setAltitudeRequired(false);
-    criteria.setSpeedRequired(false);
-
-    switch (accuracy) {
-      case lowest:
-        criteria.setAccuracy(Criteria.NO_REQUIREMENT);
-        criteria.setHorizontalAccuracy(Criteria.NO_REQUIREMENT);
-        criteria.setPowerRequirement(Criteria.NO_REQUIREMENT);
-        break;
-      case low:
-        criteria.setAccuracy(Criteria.ACCURACY_COARSE);
-        criteria.setHorizontalAccuracy(Criteria.ACCURACY_LOW);
-        criteria.setPowerRequirement(Criteria.NO_REQUIREMENT);
-        break;
-      case medium:
-        criteria.setAccuracy(Criteria.ACCURACY_COARSE);
-        criteria.setHorizontalAccuracy(Criteria.ACCURACY_MEDIUM);
-        criteria.setPowerRequirement(Criteria.POWER_MEDIUM);
-        break;
-      default:
-        criteria.setAccuracy(Criteria.ACCURACY_FINE);
-        criteria.setHorizontalAccuracy(Criteria.ACCURACY_HIGH);
-        criteria.setPowerRequirement(Criteria.POWER_HIGH);
-        break;
-    }
-
-    String provider = locationManager.getBestProvider(criteria, true);
-
-    if (provider.trim().isEmpty()) {
-      List<String> providers = locationManager.getProviders(true);
-      if (providers.size() > 0) provider = providers.get(0);
-    }
+  private static @Nullable String getBestProvider(@NonNull LocationManager locationManager) {
```

Comment:
I think we should pass in the `LocationAccuracy` as well and if the supplied value equals `LocationAccuracy.lowest` immediately return `LocationManager.PASSIVE_PROVIDER`:

```suggestion
  private static @Nullable String getBestProvider(@NonNull LocationManager locationManager, @NonNull LocationAccuracy accuracy) {
      if (accuracy == LocationAccuracy.lowest) {
          return LocationManager.PASSIVE_PROVIDER;
      }
      
```

Reply:
I agree with using the `PASSIVE_PROVIDER` on `lowest`. I did not put the logic in this method, as I feel it does not belong there. Instead, I added it to `startPositionUpdates`.

Reply:
I think it should be in this method, there should only be one place which determines the provider. This will ensure we always use the same logic when we need to determine the provider on multiple places. 

Maybe we should however rename the method as I agree that the name `getBestProvider` doesn't fit well with the `PASSIVE_PROVIDER`. So lets rename it to `getProvider` or `determineProvider`.